tendermint/src/genesis.rs

```diff
@@ -24,25 +25,9 @@ pub struct Genesis<AppState = serde_json::Value> {
     pub validators: Vec<validator::Info>,
 
     /// App hash
-    #[serde(skip_serializing_if = "Vec::is_empty", with = "serde_bytes")]
-    pub app_hash: Vec<u8>,
+    pub app_hash: String,
```

Comment:
What's your reasoning behind this change?

Reply:
In our app, we're updating the genesis.json generated from Tendermint Go. It sets `"app_hash": ""`, which is parsed as an empty `Vec`, but when we serialize it back to json, it omits the field completely. Then on subsequent run, the parser fails due to `Error("missing field `app_hash`", ...)`

Reply:
And what if you changed it to:

```suggestion
    #[serde(with = "serde_bytes")]
    pub app_hash: Vec<u8>,
```

In the Go version of Tendermint it's modeled as `HexBytes`, which is an array of bytes, and I think it should be modeled similarly here.

Reply:
unfortunately, this writes back `"app_hash": []` into JSON, which is not accepted :/

Reply:
perhaps this could use a custom serialize/deserialize with hex encoding/decoding

Reply:
`#[serde(with="serializers::bytes::hexstring")]` does it, I'll fix it up